test/common/PrepDataFuncs.cpp

```diff
@@ -341,6 +342,50 @@ namespace RcclUnitTesting
     return TEST_SUCCESS;
   }
 
+  ErrCode DefaultPrepData_AllToAllv(CollectiveArgs &collArgs)
+  {
+
+    CHECK_CALL(CheckAllocation(collArgs));
+    size_t const numInputBytes = collArgs.numInputElements * DataTypeToBytes(collArgs.dataType);
+    size_t const numOutputBytes = collArgs.numOutputElements * DataTypeToBytes(collArgs.dataType);
+
+    // calculating maxBytes as the maximum number of input bytes out of all the ranks
+    size_t maxBytes = 0;
+    for (int baseRank = 0; baseRank < collArgs.totalRanks; ++baseRank)
+    for (int offsetRank = 0; offsetRank < collArgs.totalRanks; ++offsetRank)
+    {
+      size_t rankSendCount = collArgs.options.sdispls[(baseRank)*collArgs.totalRanks+offsetRank] + collArgs.options.sendcounts[(baseRank)*collArgs.totalRanks+offsetRank];
+      if (maxBytes < rankSendCount)
+        maxBytes = rankSendCount;
+    }
+    maxBytes = maxBytes*DataTypeToBytes(collArgs.dataType);
```

Comment:
Just use *=

Reply:
Looks like you mostly use maxNumElements in loop.  Just carry that instead, and use * dataType size when allocating tempInput.